src/geo/projection/globe.js

```diff
@@ -0,0 +1,496 @@
+        const lng = lngFromMercatorX(x);
+        const lat = latFromMercatorY(y);
+        return new LngLat(lng, lat);
+    },
+
+    projectTilePoint(x: number, y: number, id: CanonicalTileID): {x: number, y: number, z: number} {
+        const tiles = Math.pow(2.0, id.z);
+        const mx = (x / EXTENT + id.x) / tiles;
+        const my = (y / EXTENT + id.y) / tiles;
+        const lat = latFromMercatorY(my);
+        const lng = lngFromMercatorX(mx);
+        const pos = latLngToECEF(lat, lng);
+
+        const bounds = globeTileBounds(id);
+        const normalizationMatrix = globeNormalizeECEF(bounds);
+        vec3.transformMat4(pos, pos, normalizationMatrix);
+
+        return {x: pos[0], y: pos[1], z: pos[2]};
+    },
+
+    locationPoint(tr: Transform, lngLat: LngLat): Point {
+        const pos = latLngToECEF(lngLat.lat, lngLat.lng);
+        const up = vec3.normalize([], pos);
+
+        const elevation = tr.elevation ?
+            tr.elevation.getAtPointOrZero(tr.locationCoordinate(lngLat), tr._centerAltitude) :
+            tr._centerAltitude;
+
+        const upScale = mercatorZfromAltitude(1, 0) * EXTENT * elevation;
+        vec3.scaleAndAdd(pos, pos, up, upScale);
+        const matrix = calculateGlobeMatrix(tr, tr.worldSize);
+        mat4.multiply(matrix, tr.pixelMatrix, matrix);
+        vec3.transformMat4(pos, pos, matrix);
+
+        return new Point(pos[0], pos[1]);
+    },
+
+    pixelsPerMeter(lat: number, worldSize: number) {
+        return mercatorZfromAltitude(1, 0) * worldSize;
+    },
+
+    createTileTransform(tr: Transform, worldSize: number): Object {
+        return new GlobeTileTransform(tr, worldSize);
+    },
+
+    farthestPixelDistance(tr: Transform): number {
+        const pixelsPerMeter = this.pixelsPerMeter(tr.center.lat, tr.worldSize);
+        const globePixelDistance = farthestPixelDistanceOnSphere(tr, pixelsPerMeter);
+        const t = globeToMercatorTransition(tr.zoom);
+        if (t > 0.0) {
+            const mercatorPixelsPerMeter = mercatorZfromAltitude(1, tr.center.lat) * tr.worldSize;
+            const mercatorPixelDistance = farthestPixelDistanceOnPlane(tr, mercatorPixelsPerMeter);
+            return interpolate(globePixelDistance, mercatorPixelDistance, t);
+        }
+        return globePixelDistance;
+    }
+};
+
+export function globeTileBounds(id: CanonicalTileID): Aabb {
+    const z = id.z;
+
+    const mn = -GLOBE_RADIUS;
+    const mx = GLOBE_RADIUS;
+
+    if (z === 0) {
+        return new Aabb([mn, mn, mn], [mx, mx, mx]);
+    } else if (z === 1) {
+        if (id.x === 0 && id.y === 0) {
+            return new Aabb([mn, mn, mn], [0, 0, mx]);
+        } else if (id.x === 1 && id.y === 0) {
+            return new Aabb([0, mn, mn], [mx, 0, mx]);
+        } else if (id.x === 0 && id.y === 1) {
+            return new Aabb([mn, 0, mn], [0, mx, mx]);
+        } else if (id.x === 1 && id.y === 1) {
+            return new Aabb([0, 0, mn], [mx, mx, mx]);
+        }
+    }
```

Comment:
These could be generated once into a lookup table.

Reply:
Good call! https://github.com/mapbox/mapbox-gl-js/pull/11329/commits/71b0bdca158902c2148840d9a67f2ad2e45dd714